boltkit/controller.py

```diff
@@ -468,23 +469,22 @@ def _install_cores(cls, path, package, core_count):
 
             config.update(core_member_home, core_config)
 
-        return initial_discovery_members
+        return initial_discovery_members, port
 
     @classmethod
-    def _install_read_replicas(cls, path, package, initial_discovery_members, core_count, read_replica_count):
-        first_bolt_port = INITIAL_BOLT_PORT + core_count
-        first_http_port = INITIAL_HTTP_PORT + core_count
-        first_https_port = INITIAL_HTTPS_PORT + core_count
-
+    def _install_read_replicas(cls, path, package, initial_discovery_members, read_replica_count, port):
         controller = _create_controller()
         for read_replica_idx in range(0, read_replica_count):
             read_replica_dir = READ_REPLICA_DIR_FORMAT % read_replica_idx
             read_replica_path = path_join(path, READ_REPLICAS_DIR, read_replica_dir)
             read_replica_home = controller.extract(package, read_replica_path)
 
-            bolt_listen_address = _localhost(first_bolt_port + read_replica_idx)
-            http_listen_address = _localhost(first_http_port + read_replica_idx)
-            https_listen_address = _localhost(first_https_port + read_replica_idx)
+            bolt_listen_address = _localhost(port)
+            port += 1
+            http_listen_address = _localhost(port)
+            port += 1
+            https_listen_address = _localhost(port)
+            port += 1
```

Comment:
An alternative to the port increment would be to use `iter`, `range` and `next`:
```
>>> i = iter(range(3))
>>> next(i)
0
>>> next(i)
1
>>> next(i)
2
```
Only downside is having to initialise the range to correct size first.